test/e2e/template.html

```diff
@@ -4,6 +4,7 @@
     <title>Playwright end-to-end test</title>
     <meta charset="utf-8" />
     <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width" />
+    <meta http-equiv="Content-Security-Policy" content="default-src 'self'; img-src *">
```

Comment:
```suggestion
    <meta
      http-equiv="Content-Security-Policy"
      content="default-src 'self'; style-src 'unsafe-inline'; img-src *"
    />
```

Webpack adds inline styles:

<img width="619" src="https://github.com/mui/mui-x/assets/13808724/d2f8bcd4-3b63-4ec7-8a9f-581f417778ff">

Also, it turns out that feature detection of eval doesn't work:

<img width="665" src="https://github.com/mui/mui-x/assets/13808724/4531d411-3fe3-4fe6-86e2-85320257b580">

Reply:
> Also, it turns out that feature detection of eval doesn't work:

Isn't this because other parts of the bundle are using eval? `try/catch` can handle CSP errors in the testing I've done.

Reply:
It's done. I've added a wrapper to repeat a test with `eval` disabled, and I've applied it to one test, which seemed enough for now.